source/common/http/conn_pool_base.h

```diff
@@ -89,6 +89,7 @@ class ConnPoolImplBase : public ConnectionPool::Instance,
     Stats::TimespanPtr conn_length_;
     Event::TimerPtr connect_timer_;
     bool resources_released_{false};
+    bool timed_out_{true};
```

Comment:
Should this be initialized false? (I don't see anything that clears, so I think every failure in onConnectionEvent will report a time out?

Reply:
gack, that was *not* supposed to be part of the commit.  That's the ASSERT-fail moral equivalent of "surely tests will fail if I set this wrong" which they surprisingly didn't.  And now we have tests that will fail :-)